packages/theme/src/generators/mui.ts

```diff
@@ -523,46 +343,79 @@ export function generateMuiTheme(mode: PaletteMode): Theme {
               color: isDarkMode ? theme.palette.text.secondary : theme.palette.secondary.light,
             },
           }),
-          arrow: ({ theme }) => ({
-            color: theme.palette.text.primary,
-          }),
+          arrow: ({ theme }) => ({ color: theme.palette.text.primary }),
         },
       },
       MuiBackdrop: {
-        styleOverrides: {
-          root: ({ theme }) => ({
-            backgroundColor: alpha(theme.palette.backdrop.main, 0.75),
-          }),
-        },
+        styleOverrides: { root: ({ theme }) => ({ backgroundColor: alpha(theme.palette.backdrop.main, 0.75) }) },
       },
       MuiSwitch: {
-        defaultProps: {
-          color: isDarkMode ? undefined : 'success',
-        },
+        defaultProps: { color: 'success' },
         styleOverrides: {
-          thumb: () => ({
-            boxShadow:
-              '0px 2px 6px -1px rgba(0, 0, 0, 0.2), 0px 1px 4px rgba(0, 0, 0, 0.14), 0px 1px 4px rgba(0, 0, 0, 0.14)',
+          root: ({ theme }) => ({
```

Comment:
This is the key area of changes. Others are lint updates only and can be ignored. 
This changes are also open here and should be resolved in that PR first
https://github.com/safe-global/safe-wallet-monorepo/pull/6915